src/app-layout/drawer/index.tsx

```diff
@@ -59,12 +47,75 @@ export function Drawer({
   onToggle,
   onClick,
   onLoseFocus,
+  drawers,
+  onResize,
+  size,
+  getMaxWidth,
+  refs,
 }: DesktopDrawerProps) {
+  const openButtonWrapperRef = useRef<HTMLElement | null>(null);
   const { TagName, iconName, getLabels } = togglesConfig[type];
   const { mainLabel, closeLabel, openLabel } = getLabels(ariaLabels);
   const drawerContentWidthOpen = isMobile ? undefined : width;
   const drawerContentWidth = isOpen ? drawerContentWidthOpen : undefined;
-  const openButtonWrapperRef = useRef<HTMLElement | null>(null);
+
+  const activeDrawer = drawers?.items.find(item => item.id === drawers.activeDrawerId);
```

Comment:
This component represents a single drawer, so finding the specific one should happen on a level above (app layout main component)

Currently this find logic runs from 1 to 4 times, depending on how many drawers are used (including the static navigation/tools ones)

Reply:
I can take a look, but may need some help moving all of this logic one level up. It's pretty entrenched right now.